test/ao/sparsity/test_scheduler.py

```diff
@@ -38,3 +38,15 @@ def test_step(self):
         assert sparsifier.module_groups[0]['sparsity_level'] == 0.5
         scheduler.step()
         assert sparsifier.module_groups[0]['sparsity_level'] == 0.25
+
+    def test_lambda_scheduler(self):
+        model = nn.Sequential(
+            nn.Linear(16, 16)
+        )
+        sparsifier = WeightNormSparsifier()
+        sparsifier.prepare(model, config=None)
+        assert sparsifier.module_groups[0]['sparsity_level'] == 0.5
+        scheduler = LambdaSL(sparsifier, lambda epoch: epoch * 10)
+        assert sparsifier.module_groups[0]['sparsity_level'] == 0.0  # Epoch 0
+        scheduler.step()
+        assert sparsifier.module_groups[0]['sparsity_level'] == 5.0  # Epoch 1
```

Comment:
Should we ensure that sparsity level is always between 0 and 1 for all schedulers?

Reply:
Currently the check is:
```
if sparsity_level >= 1.0:
    return torch.zeros(...)
elif sparsity_level <= 0.0:
    return torch.ones(...)
else:
    return weight * mask
```

I can add explicit checks to communicate to the user -- will be part of the base_sparsifier PR